src/mlpack/core/data/load_image_impl.hpp

```diff
@@ -0,0 +1,202 @@
+/**
+ * @file load_image_impl.hpp
+ * @author Mehul Kumar Nirala
+ *
+ * An image loading utillity
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_CORE_DATA_LOAD_IMAGE_IMPL_HPP
+#define MLPACK_CORE_DATA_LOAD_IMAGE_IMPL_HPP
+
+// In case it hasn't been included yet.
+#include "load_image.hpp"
+
+namespace mlpack {
+namespace data {
+
+LoadImage::LoadImage():
+        matrixWidth(-1),
```

Comment:
If we switch to ``size_t`` we could use ``0`` here to get the same functionality.

Reply:
But stb require interger pointers

Reply:
What I like to achieve here is to get a similar interface for the image loading feature as used for the rest of the codebase. If you like I can see if I can get it to work with ``size_t``.